pkg/domain/infra/abi/images.go

```diff
@@ -55,16 +64,19 @@ func (ir *ImageEngine) Prune(ctx context.Context, opts entities.ImagePruneOption
 	}
 
 	if !opts.All {
-		pruneOptions.Filters = append(pruneOptions.Filters, "dangling=true")
+		if !filterPrefix("dangling=", pruneOptions.Filters) {
```

Comment:
This the same check where @Luap99 raised concern before. If this is handled by `c/common` then we might not needed it. But I don't think there is any blocker if we are checking for duplicates.

so LGTM

Reply:
@vrothberg @Luap99 ^

Reply:
Yes, the dangling-logic can/should be removed.

Reply:
The dangling check is looking for dupicates.  This function can be called in multiple different ways.  And certain code paths and Remote API can pass in a filter with dangling=true already set, which ends up adding the dangling=true, which I am trying to prevent.  If you are suggesting the common defaults to dangling=true?

Reply:
The code in question was added to work around the duplicate checks in c/common. In the meantime, c/common accepts duplicates as long as they use the same value, so we don't need the custom check here anymore.